src/include/sof/lib/dma.h

```diff
@@ -564,19 +570,56 @@ static inline bool dma_is_scheduling_source(struct dma_chan_data *channel)
 	return ret;
 }
 
-static inline void dma_sg_init(struct dma_sg_elem_array *ea)
+static inline void dma_sg_init(struct dma_sg_array *sg_array)
+{
+	sg_array->count = 0;
+	sg_array->elems = NULL;
+}
+
+static inline void dma_sg_elems_init(struct dma_sg_elem_array *elem_array)
 {
-	ea->count = 0;
-	ea->elems = NULL;
+	elem_array->count = 0;
+	elem_array->elems = NULL;
 }
 
+static inline void dma_sg_elem_trace(struct dma_sg_elem *sg_elem)
+{
+	trace_event(TRACE_CLASS_DMA, "dma_sg_elem_trace() src %X dest %X size %d",
+		    sg_elem->src, sg_elem->dest, sg_elem->size);
```

Comment:
Since this is probably for debug, we should do tracev_event instead of trace_event.